controllers/restaurantsChainController.js

```diff
@@ -0,0 +1,102 @@
+import { httpStatus } from '../utils/httpStatusCode.js';
+import { restaurantsChainService } from '../services/restaurantsChainService.js';
+
+const getChains = async function (req, res) {
+  try {
+    const chain = await restaurantsChainService.getChains(req.query);
+
+    if (chain.length === 0) {
+      return res.status(httpStatus.NO_CONTENT).json({
+        message: 'No chain found',
+      });
+    }
+
+    return res.status(httpStatus.OK).json({
+      success: true,
+      data: chain,
+    });
+  } catch (err) {
+    return res.status(httpStatus.INTERNAL_SERVER_ERROR).json({
+      success: false,
+      error: 'Server Error',
+    });
+  }
+};
+
+const addChain = async function (req, res) {
+  try {
+    const chain = await restaurantsChainService.addChain(req.body);
+
+    return res.status(httpStatus.CREATED).json({
+      success: true,
+      data: chain,
+    });
+  } catch (err) {
+    return res.status(httpStatus.INTERNAL_SERVER_ERROR).json({
+      success: false,
+      error: 'Server Error',
+    });
+  }
+};
+
+const getChainById = async function (req, res) {
+  try {
+    const chain = await restaurantsChainService.getChainById(req.params);
+    if (!chain) {
+      return res.status(httpStatus.NO_CONTENT).json({
+        message: 'ID not found.',
```

Comment:
I would return 404, id is not found in db and that is an error

Reply:
done